assets/app.js

```diff
@@ -579,17 +583,20 @@ new Vue({
             }
         },
         translatorComponents: function(){
-            if (ComponentOptions && ComponentOptions.content) {
-                var response = []
-                ComponentOptions.content.forEach(function (option, index) {
-                    if(option.name !== '-' && option.is_active==1 && option.translation_requested == 1){
-                        response.push(option);
-                    }
-                })
-                return response
-            }
-            else {
-                return []
+            if(!this.reloadComponents){
+                this.reloadComponents = true
+                if (ComponentOptions && ComponentOptions.content) {
+                    var response = []
+                    ComponentOptions.content.forEach(function (option, index) {
+                        if(option.name !== '-' && option.is_active==1 && option.translation_requested == 1){
+                            response.push(option);
+                        }
+                    })
+                    return response
+                }
+                else {
+                    return []
```

Comment:
you return an empty array here if `reloadComponents===true` but nothing if `translatorBaseProducts===true`
Both approaches should be the same or not?. Where do we load them from the api? The api-call should be more complicated since you have to handle two different data-types in the case we already read one of them. (translatorComponents becomes [] and translatorBaseProduct becomes null).